### ISSUE TITLE
Convert DuoTone Themes to Base16 Schemes

### ISSUE THREAD

**Issue by aloisdg** [alexbooker/base16-builder#49]
Hello,

There are a lot of very nice theme for the Atom IDE. I would love to try them somewhere else. Can we think about a way to use them as scheme for base16-buider?

You can discover the concept [here](http://simurai.com/projects/2016/01/01/duotone-themes) (by @simurai).

**Comment by alexbooker** [comment]
Definitely something to consider in the future :smile:

**Comment by simurai** [comment]
I think @atelierbram used base16-builder to generate these: http://atelierbram.github.io/syntax-highlighting/duotones/

**Comment by aloisdg** [comment]
@simurai Thank you for the comment. @atelierbram made a great job. It will help a lot!

**Comment by aloisdg** [comment]
@simurai Just diff a bit and found [this](https://github.com/atelierbram/syntax-highlighting/tree/master/duotones/schemes). PR is on the way!

**Comment by atelierbram** [comment]
@aloisdg Thanks for the compliments, it's flattering; but a few cautionary words on the PR which I hope will be helpful. The proof will be in trying to make these into light background themes, which can be an interesting academic exercise, but until now I only have been bothered with the dark background templates. It could be a tricky project trying to reconcile the differences in philosophy between DuoTone Themes and the Base16 themes, and to be honest, I question if it is doable without compromising one or the other, and in art there is no room for compromises.  
 
So what I felt I had to do was to adept the (dark-background-)templates for Vim, Sublime Text, Prism and CodeMirror (_the only "finished" themes_) in order to get near the original, using the base16 template system and executable bash script as a convenient building tool. But which admittedly also leads me to explain my position in this issue here. 
 
[Here are some demos](http://atelierbram.github.io/syntax-highlighting/duotones/demo/dark-space/) for Prism which can make the mapping of the color-values to base16-variables I used insightful. But, like I wrote, although bringing those schemes in here would probably get you more than halfway there (for _most applications_, but IMO not _all the way_ there for _all_ existing templates (for they where created with a different blueprint in mind).

**Comment by atelierbram** [comment]
@alexbooker @aloisdg :+1:  Love the new CLI; having played with it I have to say I am impressed by it's versatility and flexibility.

An update on my efforts of converting DuoTone Themes in light and dark variations. Bad news: wasn't really doable IMO with 16 color-variables (without it being a tangled mess), so no PR here. Good news: it still can be done with more color variables _and_ generated with Base16 Builder. A proof of concept [here for Prism](https://github.com/atelierbram/duotones-prism) with 32 colorvalues (not all used), but I hope it makes sense when seeing [this demo](http://atelierbram.github.io/duotones-prism/demo/evening/dark/). Am planning to take this further, for I think this has great potential, and wouldn't mind any feedback.

**Comment by aloisdg** [comment]
Thank you for your answer and the demo!

You Prism demo is, at least for me, completely usable. Great works you made their. If you can go further, I would love to see this :)

Just to be sure, you are not using the same color (or same colors with different shades) in both dark and light mode, aren't you?

**Comment by alexbooker** [comment]
@atelierbram Yes, thank you! I, too, would be very interested in seeing this. Good work so far :smile: :ok_hand:

**Comment by atelierbram** [comment]
> Just to be sure, you are not using the same color (or same colors with different shades) in both dark and light mode, aren't you?

@aloisdg I guess I am, that is to say; the colors for the light mode map to different colorvalues but from the same `scheme/colorscheme.yml` file with 32 colorvalues declared (_maybe could be 24, but this gives a bit more flexibility_). But I hope hitting the "show colornames" button on [the demo](http://atelierbram.github.io/duotones-prism/demo/desert/dark/) and switching between dark - and light mode will make this a bit more concrete and visual.

Maybe it's fair to say that my mental model of Simurai's great concept is (schematicaly) a bit like this: two colors with each two tonal scales ranging from dark to light in a saturated and de-saturated version. So objectively you have _two_ hue values, but perceptively _four_ colors in a dark to light scale. It's genius in it's simplicity for it almost guarantees harmonious colorschemes while at the same time giving just enough variation to make it visually interesting.

@alexbooker Have been thinking on calling it base32-duotones or duotones-base32 or something, but am undecided about it. On the one hand I want to avoid the confusion with the original project from @simurai , and confusion with base16 implementations, on the other hand, although this has great potential, I don't want to get overboard with this just yet, so yeah, need to think about that.

**Comment by aloisdg** [comment]
@atelierbram

I don't think I have problem with the idea of splitting schemes by brightness. We can apply the same model used for template. We would have `db/schemes/{name}/{brightness}.yml`. Most of the old schemes would be equals in both dark and light version (lot of duplicate here but it is the current choice made for template). What do you think of this alternative?

**Comment by atelierbram** [comment]
Will give it a day to wrap my mind around this idea. Right now, I think this will overcomplicate the base16 concept, and take away some clarity for Duotones, for it means two YML scheme files: `db/schemes/{name}/dark.yml` and `db/schemes/{name}/light.yml`, no? Somehow this doesn't sit quiet right with me yet ... what does @alexbooker think?

**Comment by aloisdg** [comment]
@atelierbram

You are right, It may not fit very well. Also It could be a full other project as you said before. (I like the name Base2 or BaseDuo ;).

**Comment by alexbooker** [comment]
@atelierbram, I value simplicity _a lot_. You are spot on when you say this _"will overcomplicate the base16 concept, and take away some clarity for Duotones"_.

This is a tricky one :confounded: 

All I can really say at this point is that I want to help you leverage the "core" as best I can. Maybe you can use the [JavaScript API](https://github.com/base16-builder/base16-builder/issues/50) to build a separate tool. Another option would be to interface with command-line itself.

**Comment by atelierbram** [comment]
Thanks for the feedback, and yes, like with many things, the more you think about it, the more it seems obvious. 

@aloisdg Your comment on naming was actually quite helpful in finding the right direction for picking something. So I settled on **Base2Tone**, which is less memorable then "Duotones", but the plural "s" to differentiate from "DuoTone Themes" was always more than a bit lame. And with "Base2" this loosely references Base16 and maybe also an association with "_a build-tool_", if this makes sense. So I'm happy enough with it, for it's also a fine excuse to create an animated SVG-logo with alternative spelling ("Base > Tone")  :smirk:

@alexbooker When I am far enough into the project to oversee the scope of it, maybe I will come back to you, that is, if - or when there will be a "_Base2Tone-Builder_" repo, _if_ this is ever needed _at all_, I'm not sure. For now that will be overkill, and the command-line interface works fine with me. So I will just point to here for the dependencies, and let people install "Base16 Builder" globally on there machine from NPM, if they want to build there own scheme or template. Having those commands in executable bash-scripts, keeps it simple, but I 'm a low-fi creature :bug:, and don't know much about JavaScript.

Separating out into different repos for different apps, (_like is being done with [Base2Tone-prism](https://github.com/atelierbram/Base2Tone-prism)_) seems to be the right move at this time. Such a horizontal setup also allows linking to those repos from the main "Base2Tone" repo; with the advantage that this will not only have to be my own repos, but also potentially those from others, which might help keeping the project maintainable. So if you ever want to create a "Base2Tone-{app-name}", you are more then welcome and know where to find me to link this up from the main repo! :wink:

**Comment by aloisdg** [comment]
@atelierbram 

Glad to be helpful for the name!

I think we can close this issue for now. What do you think @alexbooker?

**Comment by alexbooker** [comment]
@atelierbram I appreciate it a lot. I'm sure both myself and @aloisdg will be glad to help with any code relating to this project. Don't hesitate to get in touch, OK :smile:?

@aloisdg I don't think there's any harm in keeping it open for a while, do you?

**Comment by aloisdg** [comment]
@alexbooker of course we can. :smile:
